api/src/main/java/org/pmiops/workbench/api/UserMetricsController.java

```diff
@@ -141,7 +144,10 @@ public void setDistinctWorkspaceLimit(int limit) {
         })
         .forEach(userRecentResource -> {
           RecentResource resource = TO_CLIENT.apply(userRecentResource);
-          resource.setPermission(workspaceAccessMap.get(userRecentResource.getWorkspaceId()));
+          WorkspaceResponse workspaceDetails = workspaceAccessMap.get(userRecentResource.getWorkspaceId());
+          resource.setPermission(workspaceDetails.getAccessLevel());
+          resource.setWorkspaceNamespace(workspaceDetails.getWorkspace().getNamespace());
+          resource.setWorkspaceFirecloudName(workspaceDetails.getWorkspace().getName());
```

Comment:
Another way to do this would be to have another `Function` object that takes a `WorkspaceResponse` and also returns a `RecentResource`, then you could chain the two together with `andThen` instead of doing all of the resource.set* calls inside the loop. No need to implement that for this PR, just wanted to point out an alternative approach.